2.0/m_banprotect.cpp

```diff
@@ -0,0 +1,114 @@
+/*
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/* $ModAuthor: Sebastian Nielsen */
+/* $ModAuthorMail: sebastian@sebbe.eu */
+/* $ModDesc: Prevents a lower rank from removing a listmode item set by a higher rank */
+/* $ModDepends: core 2.0 */
+
+#include "inspircd.h"
+
+class Banprotector
+{
+ public:
+	std::map<std::string, unsigned int> banrank;
+	Banprotector() {}
+
+	void addrank(const char& modeparam, const std::string& banparam, const unsigned int& rank)
+	{
+		std::string fullban = ConvToStr(modeparam) + "%" + ConvToStr(banparam);
```

Comment:
Fixed. The reason I used a concatenated string instead of map of maps, was that it was both easier and also, I Think it would be more efficient to store a string->rank map instead of a char->string->rank map. It might be a performance increase to use a map of maps as the listmode only needs to be saved once for each banlist, but it depends on how a std::map is saved physically into memory, if the inner maps uses Control codes to delimit the values, it could aswell be same or heavier than using concatenated key strings.

However, as you requested, I have changed into a map of maps instead.